docs/src/tutorial.md

```diff
@@ -139,3 +139,172 @@ sol = WorldDynamics.solve(system, (1900, 2100))
+
+Finally, the model can be solved and simulated by using the `solve` and the `plotvariables` functions that we already used above. In particular, the file `plots.jl` (we assume that also this file is contained in the directory `nonrenewablestock`) does it in order to reproduce Figure 3.9 of the chapter of the above mentioned book.
+
+```
+using ModelingToolkit
+using DifferentialEquations
+
+function nrs_run_solution()
+    isdefined(@__MODULE__, :_solution_nrs_run) && return _solution_nrs_run
+    global _solution_nrs_run = solve(nrs_run(), (0, 200), solver=Tsit5(), dt=0.015625, dtmax=0.015625)
+    return _solution_nrs_run
+end
+function _variables_nrs()
+    @named nrs = non_renewable_stock()
+    variables = [
+        (nrs.capital, 0, 30, "Capital"),
+        (nrs.extraction, 0, 15, "Extraction"),
+        (nrs.investment, 0, 2, "Investment"),
+        (nrs.depreciation, 0, 2, "Investment"),
+        (nrs.resource, 0, 1000, "Resource"),
+    ]
+    return variables
+end
+
+fig_3_9(; kwargs...) = plotvariables(nrs_run_solution(), (t, 0, 200), _variables_nrs(); title="Simulation output showing stocks and flows", showaxis=false, showlegend=true, kwargs...)
+```
+Note that, for performance reasons, the definition of the function `nrs_run_solution` starts by checking whether the solution of the model is already available: in this case, nothing is done.
+
+### Integrating the model into WorldDynamics
+
+In order to integrate the new model in the `WorldDynamics` framework, we can now define a Julia module `Duggan.jl` as follows (we assume that this source file is contained in the directory `Duggan`).
+
+```
+module Duggan
+using ModelingToolkit
+using WorldDynamics
+include("NonRenewableStock.jl")
+end
+```
+The file `NonRenewableStock.jl` (we assume that also this file is contained in the directory `Duggan`) simply includes all the Julia source files we have written above.
+
+```
+module NonRenewableStock
+using WorldDynamics
+using ModelingToolkit
+include("nonrenewablestock/tables.jl")
+include("nonrenewablestock/parameters.jl")
+include("nonrenewablestock/initialisations.jl")
+include("nonrenewablestock/subsystems.jl")
+include("nonrenewablestock/scenarios.jl")
+include("nonrenewablestock/plots.jl")
+end
+```
+
+### Solving the model and producing the figure
+
+We assume that we execute the Julia REPL from the directory containing the folder `Duggan`. We can solve the model and produce the desired figure by simply executing the following two instructions.
+
+```
+include("Duggan/Duggan.jl")
+Main.Duggan.NonRenewableStock.fig_3_9()
```

Comment:
Do we need `Main`?

Reply:
As far as I know, yes. But I am not a Julia expert...